Makefile

```
@@ -1,25 +1,49 @@
-
 .PHONY: all deploy build helm
 .ONESHELL: test
 
 ifeq ($(K8S_VERSION),)
-K8S_VERSION := v1.22.13
+K8S_VERSION := v1.28.4
 endif
 
+LOCALBIN_DIR ?= $(shell pwd)/bin
+$(LOCALBIN_DIR):
+	mkdir -p $(LOCALBIN_DIR)
+
 k3d:
 	@curl -s https://raw.githubusercontent.com/k3d-io/k3d/main/install.sh | bash
 	@k3d cluster create myk3s -i rancher/k3s:$(K8S_VERSION)-k3s1
 	@kubectl get pod
 
-lint: ## lint helm manifests
-	@helm lint -f charts/db-operator/values.yaml -f charts/db-operator/ci/ci-1-values.yaml --strict ./charts/db-operator
-	@helm lint -f charts/db-instances/values.yaml --strict ./charts/db-instances
+helmfile-operator: ## Syncs the helmfile with dependencies for testing operator
+	helmfile sync -f ./helmfile.yaml
+
+helmfile-instances: ## Syncs the helmfile with dependencies for testing dbinstances
+	helmfile sync -f ./helmfile.yaml -e instances
+
+lint: ## lint helm charts
+	ct lint --target-branch main --validate-maintainers=false
+
+test-values: 
+	cd charts/db-operator && \
+		bash ./scripts/test_values -p ./ci/unit-test
+
+test-operator: helmfile-operator ## test helm charts
+	ct install --target-branch main --charts ./charts/db-operator
+
+test-upgrade-operator: helmfile-operator ## test helm charts
+	ct install --upgrade --target-branch main --charts ./charts/db-operator
+
+test-instances: helmfile-instances ## test helm charts
+	ct install --target-branch main --charts ./charts/db-instances
+
+test-upgrade-instances: helmfile-instances ## test helm charts
+	ct install --upgrade --target-branch main --charts ./charts/db-instances
+
+.PHONY: gen_docs
+gen-docs: ## Generate helm documentation
+	test -s $(LOCALBIN)/setup-envtest || GOBIN=$(LOCALBIN_DIR) go install github.com/norwoodj/helm-docs/cmd/helm-docs@latest
+	./bin/helm-docs --template-files=./charts/db-operator/README.md.gotmpl \
+		--sort-values-order file --chart-to-generate=charts/db-operator
+	./bin/helm-docs --template-files=./charts/db-isntances/README.md.gotmpl --sort-values-order file --chart-to-generate=charts/db-instances
```

Wouldn't this be nicer with a loop? I know two charts is not much right now, but it would remove the need to make changes here, if a new chart is added.
```suggestion
	CHARTS_DIR = ./charts
	$(foreach chart_dir, $(wildcard $(CHARTS_DIR)/*), 
		./bin/helm-docs --template-files=$(chart_dir)/README.md.gotmpl \
			--sort-values-order file --chart-to-generate=$(chart_dir)
	;)
```

---

So, though I like the idea of decreasing copy-paste, we don't think it really makes sense for us to add a loop here, at least while we have only two charts. I find this code a way harder to read than the one with copy-paste, so let's way until we have a third chart

---

Ok. Fair enough. I understand your reasoning.